CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinkInventoryBase.h

```diff
@@ -0,0 +1,233 @@
+*                             Indiana                                   *
+*                                                                       *
+* This program is free software; IF YOU AGREE TO CITE USE OF CompuCell  *
+*  IN ALL RELATED RESEARCH PUBLICATIONS according to the terms of the   *
+*  CompuCell GNU General Public License RIDER you can redistribute it   *
+* and/or modify it under the terms of the GNU General Public License as *
+*  published by the Free Software Foundation; either version 2 of the   *
+*         License, or (at your option) any later version.               *
+*                                                                       *
+* This program is distributed in the hope that it will be useful, but   *
+*      WITHOUT ANY WARRANTY; without even the implied warranty of       *
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU    *
+*             General Public License for more details.                  *
+*                                                                       *
+*  You should have received a copy of the GNU General Public License    *
+*     along with this program; if not, write to the Free Software       *
+*      Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.        *
+*************************************************************************/
+#ifndef FOCALPOINTPLASTICITYLINKINVENTORYBASE_H
+#define FOCALPOINTPLASTICITYLINKINVENTORYBASE_H
+
+#include <map>
+#include <set>
+#include <vector>
+
+#include <CompuCell3D/CC3D.h>
+
+#include "FocalPointPlasticityLinks.h"
+#include "FocalPointPlasticityDLLSpecifier.h"
+
+namespace CompuCell3D {
+
+	/**
+	Written by T.J. Sego, Ph.D.
+	*/
+
+	class Potts3D;
+	class FocalPointPlasticityTracker;
+
+	typedef std::set<FocalPointPlasticityTrackerData> FPPTrackerDataSet;
+
+	template <typename T>
+	class FPPLinkListBase : public std::vector<T*> {
+	public:
+		typedef typename std::vector<T*>::iterator FPPLinkListIterator_t;
+		virtual ~FPPLinkListBase() {}
+	};
+
+	class FOCALPOINTPLASTICITY_EXPORT FPPLinkID {
+	public:
+		long id0;
+		long id1;
+
+		FPPLinkID() : id0(0), id1(0) {};
+		FPPLinkID(long _id0, long _id1)
+		{
+			if (_id0 < _id1) {
+				id0 = _id0;
+				id1 = _id1;
+			}
+			else {
+				id1 = _id0;
+				id0 = _id1;
+			}
+		}
+		~FPPLinkID() {}
+		bool operator < (const FPPLinkID & _rhs) const { return id0 < _rhs.id0 || (id0 == _rhs.id0 && id1 < _rhs.id1); }
+		bool operator == (const FPPLinkID & _rhs) const { return id0 == _rhs.id0 && id1 == _rhs.id1; }
+		bool operator != (const FPPLinkID & _rhs) const { return !(operator==(_rhs)); }
+	};
+
+	template <class LinkType> class FPPLinkInventoryBase;
+
+	template <class LinkType>
+	class FOCALPOINTPLASTICITY_EXPORT FPPLinkInventoryTracker {
+	public:
+
+		FPPLinkInventoryTracker() {};
+		~FPPLinkInventoryTracker() {};
+		
+		FPPLinkInventoryBase<LinkType> linkInv;
+	};
+
+
+	template <class LinkType>
+	class FOCALPOINTPLASTICITY_EXPORT FPPLinkInventoryBase {
```

Comment:
If possible I would create just one Inventory class that handles all link types I really dont' think there is a need to have template and a base class for relatively simple concept like inventory of links. In a sense inventory of links is a container that holds objects of type link. And all links are of the same type (with anchor being tiny exception)

Reply:
See my commentary above about the approach adopted here. In general, I agree that simpler is better.